Sources/SwiftUICoordinator/Navigator/Navigator.swift

```diff
@@ -38,25 +42,28 @@ public extension Navigator where Self: Coordinator, Self: RouterViewFactory {
     var visibleViewController: UIViewController? {
         return navigationController.visibleViewController
     }
+    
+    // MARK: - Public methods
 
     func start() throws {
         guard let route = startRoute else {
-            return
+            throw NavigatorError.startRouteMissing
         }
         
         try show(route: route)
     }
 
     func show(route: Route) throws {
         let view = self.view(for: route)
-            .ifLet(route.title) { view, value in
-                view.navigationTitle(value)
-            }
-
+            .navigationTitle(route.title ?? "")
```

Comment:
I ran some tests and with this change, if the `NavigationRoute.title` property is nil, an empty string will be set as the title of the corresponding ViewController. This means that when pushing another controller on the stack, the back button will have an empty string instead of the system `Back` text.
Not sure if this is intentional or not? 
<img width="278" alt="Screenshot 2023-05-19 at 7 08 58 AM" src="https://github.com/erikdrobne/SwiftUICoordinator/assets/44533/897074b4-3141-45ac-a943-e6c078328667">

Reply:
You're right, needs to be resolved before merging.